pkg/render/kube-controllers.go

```diff
@@ -30,23 +30,30 @@ import (
 
 var replicas int32 = 1
 
-func KubeControllers(cr *operator.Installation) *kubeControllersComponent {
+func KubeControllers(cr *operator.Installation, managerSecret *v1.Secret) *kubeControllersComponent {
 	return &kubeControllersComponent{
 		cr: cr,
+		managerSecret: managerSecret,
 	}
 }
 
 type kubeControllersComponent struct {
 	cr *operator.Installation
+	managerSecret *v1.Secret
 }
 
 func (c *kubeControllersComponent) Objects() ([]runtime.Object, []runtime.Object) {
-	return []runtime.Object{
+	kubeControllerObjects := []runtime.Object{
 		c.controllersServiceAccount(),
 		c.controllersRole(),
 		c.controllersRoleBinding(),
 		c.controllersDeployment(),
-	}, nil
+	}
+	if c.managerSecret != nil {
```

Comment:
It doesn't go here but in render/kube-controllers.go but with this same case check we need to add an AnnotationHash to the deployment to make sure the deployment will be updated/restarted when this secret is CRUD'ed.

Reply:
We might need to sync up on this comment. I think the comment might have been needed on a different file than `render/kube_controllers.go` ?

Reply:
You've done what I wanted with the kubeControllerAnnotations